### PR TITLE
fix(notify-sync): Do not lock before the select

### PR DESCRIPTION
**Pull request by lvalerom**
## Description

Do not lock before the select.

## Checklist
- [ ] Investigated and inspected CI test results
- ~[ ] Unit test and regression tests added~
- ~[ ] Evaluated and added CHANGELOG entry if required~
- ~[ ] Determined and documented upgrade steps~
- ~[ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))~

If any of these don't apply, please comment below.

## Testing Performed

- [ ] CI

### Reminder for reviewers

In addition to reviewing code here, reviewers **must** also review testing and request further testing in case the
performed one does not seem sufficient. As a reviewer, you must not approve the change until you understand the
performed testing and you are satisfied with it.

### CODE REVIEW

File: sensor/common/sensor/sensor.go [hunk 1]
```diff
@@ -410,10 +410,11 @@ func (s *Sensor) communicationWithCentralWithRetries(centralReachable *concurren
...
 		// Try to create a central communication component. This component will fail (Stopped() signal) if the connection
 		// suddenly broke.
-		s.centralCommunication = NewCentralCommunication(s.reconnect.Load(), s.reconcile.Load(), s.components...)
+		centralCommunication := NewCentralCommunication(s.reconnect.Load(), s.reconcile.Load(), s.components...)
 		syncDone := concurrency.NewSignal()
-		s.centralCommunication.Start(central.NewSensorServiceClient(s.centralConnection), centralReachable, &syncDone, s.configHandler, s.detector)
-		go s.notifySyncDone(&syncDone)
+		centralCommunication.Start(central.NewSensorServiceClient(s.centralConnection), centralReachable, &syncDone, s.configHandler, s.detector)
+		go s.notifySyncDone(&syncDone, centralCommunication)
+		s.centralCommunication = centralCommunication
```
Comment (vikin91) [inline sensor/common/sensor/sensor.go:417]: Wouldn't that be safer?

```suggestion
		s.centralCommunication = centralCommunication
		go s.notifySyncDone(&syncDone, centralCommunication)
```

**Review by vikin91 — APPROVED** [review]
Looks good! I would be curious if that fixes the issue from ROX-20338

File: sensor/common/sensor/sensor.go (hunk 1 above)
Comment (lvalerom) [reply sensor/common/sensor/sensor.go]: Good catch! I moved it one line above (before calling `Start) since `Start` spawns a goroutine.

**Review by fredrb — APPROVED** [review]

**Comment by vikin91** [comment]
CI all green - brilliant! Now, we do not need https://github.com/stackrox/stackrox/pull/9089 anymore (it should disable the test, but apparently, the disabled test failed 😂 )

**Pull request merged**